keras_core/layers/preprocessing/random_flip.py

```diff
@@ -46,39 +42,54 @@ class RandomFlip(Layer):
     def __init__(
         self, mode=HORIZONTAL_AND_VERTICAL, seed=None, name=None, **kwargs
     ):
-        if not tf.available:
-            raise ImportError(
-                "Layer RandomFlip requires TensorFlow. "
-                "Install it via `pip install tensorflow`."
-            )
-
         super().__init__(name=name)
-        self.seed = seed or backend.random.make_default_seed()
-        self.layer = tf.keras.layers.RandomFlip(
-            mode=mode,
-            name=name,
-            seed=self.seed,
-            **kwargs,
-        )
+        self.seed = seed
+        self.generator = SeedGenerator(seed)
+        self.mode = mode
         self.supports_jit = False
         self._convert_input_args = False
         self._allow_non_tensor_positional_args = True
 
+    @property
+    def horizontal(self) -> bool:
+        return self.mode == HORIZONTAL or self.mode == HORIZONTAL_AND_VERTICAL
+
+    @property
+    def vertical(self) -> bool:
+        return self.mode == VERTICAL or self.mode == HORIZONTAL_AND_VERTICAL
+
     def call(self, inputs, training=True):
-        if not isinstance(inputs, (tf.Tensor, np.ndarray, list, tuple)):
-            inputs = tf.convert_to_tensor(backend.convert_to_numpy(inputs))
-        outputs = self.layer.call(inputs, training=training)
-        if (
-            backend.backend() != "tensorflow"
-            and not backend_utils.in_tf_graph()
-        ):
-            outputs = backend.convert_to_tensor(outputs)
-        return outputs
+        def random_flipped_inputs(inputs):
+            flipped_outputs = inputs
+            seed_generator = self._get_seed_generator()
+            if self.horizontal:
+                if (
```

Comment:
You can't use `if` blocks, since that won't work with compilation. You should use an `ops.cond` statement.